File: data/multilang_comparison/program_julia/program_julia/src/func28.jl
include("program_julia_head.jl")
function func28(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   if (PATH0 & 1) != 0
      params0 = JArray_param{Int}()
      params0.size = 0
      array7 = func29(params0, get_path(), loopsFactor);
      array7.refC -= 1
      if array7.refC == 0
          empty!(array7.data)
          array7 = nothing
      end
   
   else 
      params0 = JArray_param{Int}()
      params0.size = 0
      array26 = func30(params0, get_path(), loopsFactor);
      array26.refC -= 1
      if array26.refC == 0
          empty!(array26.data)
          array26 = nothing
      end
   end
   params0 = JArray_param{Int}()
   params0.size = 0
   array46 = func31(params0, get_path(), loopsFactor);
   return array46;
end

File: data/multilang_comparison/program_go/program_go/src/func28.go
package main
import "math/rand"
var _ = rand.Intn
func Func28(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   if (PATH0 & 1) != 0 {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array7 *Array = Func29(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array7.RefC--
      if array7.RefC == 0 {
          array7.Data = nil
          array7 = nil
      }
   
   } else {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array26 *Array = Func30(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array26.RefC--
      if array26.RefC == 0 {
          array26.Data = nil
          array26 = nil
      }
   }
   var params0 ArrayParam
   params0.Size = 0
   params0.Data = make([]*Array, params0.Size)
   var array46 *Array = Func31(&params0, getPath(), loopsFactor);
   params0.Data = nil
   params0.Size = 0
   return array46;
}

